### ISSUE TITLE
Click sub-command name renaming not detected

### ISSUE THREAD

**Issue by Querela** [click-contrib/sphinx-click#74]
Hi, I'm using the most current version of **Sphinx** (3.4.2), **Sphinx-Click** (latest) and **click** (7.1.2).

My situation is like this:
- I have a separate package with a click cli, which has a group named `main` (module `modA.cli` in package packageA)
- in another package (module `modB.cli` in package packageB) I also have a CLI and import the group `main` from `modA.cli` as `submain`
- I add the `submain` group to the `modB.cli` command groups as a subcommand.
If I now execute `packageB` I will see the subcommand `submain` as possible choice. In the generated documentation it however will still appear as `main` (name from `modA.cli` I would suppose)

Not sure if the example code can reproduce this or if it has to be separate packages (not just modules)...
```python
# packageA modA.cli

import click

@click.group()
def main():
    pass


@main.group()
def subA():
    pass
```
```python
# packageB modB.cli

import click
from modA.cli import main as submain

@click.group()
def main():
    pass

main.add_command(submain, "submain")  # <-- here I add the commandgroup from packageB and rename the command
# without renaming here, it will show up as `main` in the CLI

@main.group()
def subB():
    pass
```
I will then call the `modB.cli:main()` entry point.

Documentation is dead simple:
```rst
.. click:: modB.cli:main
  :prog: packageB
  :nested: full
```

**Comment by stephenfin** [comment]
Yeah, this isn't something we currently support. I'm not sure how easy it would be to fix this since we don't fully evaluate the application.

**Comment by Querela** [comment]
Ok. That makes sense a guards against some side effects.
Is it possible to manually rename the command in the docs? (Also complicated probably because of any possible depth.)

**Comment by stephenfin** [comment]
Unfortunately not, no

**Comment by Querela** [comment]
Hmm.
_It's an internal project. And documentation is currently not done automatically. So maybe just manual postprocessing to update the 5-6 occurrences where it is wrong. Or just a notice somewhere._
I can't think of any easy solution/fix for now ...

**Comment by wvxvw** [comment]
It's even worse actually. If you simply move code around, for example, if you split your code into multiple modules, the documentation will silently generate something random (something that looks like the previous version of CLI, before it was moved into another module). Running Sphinx with `-Ea`, deleting the output directory doesn't help. The documentation generated is simply incorrect, and there's no notification about not being able to generate the correct one.

At the very least, please throw an error. I've only accidentally discovered that the documentation is broken after it's been broken like this for a long time.

**Comment by kaczmarj** [comment]
the renamed names are lost in these lines: https://github.com/click-contrib/sphinx-click/blob/aa0d5b7bcda183d20c7b98a9a1c9acf1e003af97/sphinx_click/ext.py#L285-L288

prior to those lines, for a `click.Group`, `lookup` is a dictionary that maps the new names to the `click.Command` objects. those lines return a list that strips away the new names. perhaps the lookup can be maintained, and the commands can be ordered alphabetically using a OrderedDict (or a list of tuples).

**Comment by stephenfin** [comment]
I'd welcome a pull request to fix this. Please include tests

**Comment by brgstfzj** [comment]
Hi there! I am interested in this fix, are there any updates on this?

**Comment by stephenfin** [comment]
Please see my earlier response above

https://github.com/click-contrib/sphinx-click/issues/74#issuecomment-1399320427
